### PR TITLE
Alpine sidecar

### PR DESCRIPTION
**Pull request by GregHanson**

### CODE REVIEW

**Comment by rshriram** [comment]
The original bookinfo app was based on `jfloff/alpine-python:2.7-onbuild` . The `-onbuild` is very convenient since we can build one base image containing sidecar + python, and then we can build any number of python apps by simply including a requirements.txt file in the application directory. It will install these in one RUN command, reducing the number of layers. In order for this to work reliably for different python apps, we need to have build-essentials and git. Especially git since people can install packages directly from git by adding lines like this in their requirements.txt.
`-e git://github.com/mozilla/elasticutils.git#egg=elasticutils`


So I suggest refactoring this PR into something like this:
1. create a base image a8-sidecar-python-alpine:2.7-onbuild
2. change the dockerfile.sidecar to import from the above base image instead of jfloff/alpine-python:2.7-onbuild

```
FROM amalgam8/a8-sidecar-python-alpine:2.7-onbuild
RUN mkdir -p /opt/microservices
COPY . /opt/microservices/
EXPOSE 9080
WORKDIR /opt/microservices
CMD python details.py 9080 http://localhost:6379
```

**Comment by GregHanson** [comment]
@rshriram I think I cover that in the docker file, with the exception of the missing "git" package:
ADD requirements.txt /opt
RUN apk add --no-cache --virtual .build-deps \
    build-base \
    && pip install -r /opt/requirements.txt \
    && apk del .build-deps

those steps do an apk add of the buildtools just for the pip install command - they are removed afterwords so as they do not contribute to the final images size.  So as of right now, they can still include whatever they want in the requirements.txt - unless I am misunderstanding your comment

**Comment by rshriram** [comment]
The `-onbuild` is dynamic. Your current dockerfile is adding the requirements.txt into the image and then installing packages. In the non-sidecar dockerfile, notice that requirements.txt is not added to the image at all. For example, https://github.com/jfloff/alpine-python/blob/master/2.7/onbuild/Dockerfile

```
FROM jfloff/alpine-python:2.7

# install requirements
# this way when you build you won't need to install again
# ans since COPY is cached we don't need to wait
ONBUILD COPY ./requirements.txt /tmp/requirements.txt
ONBUILD RUN pip install -r /tmp/requirements.txt
```

This is the base image for 2.7-onbuild. It uses Docker's ONBUILD command to pick up dependencies from the source folder, only during image build.

For the moment though, it seems premature to optimize the build process just for the bookinfo and helloworld apps. So lets just go with this.

**Comment by rshriram** [comment]
What about other files that use the bookinfo/helloworld images with embedded sidecar? like rest of dockercompose and marathon? and the kubernetes files that are also using older ubuntu based sidecars (0.1)?

File: docker/gateway.yaml [hunk 1]
```diff
@@ -14,7 +14,7 @@
 
 ##API Gateway##
 gateway:
-  image: amalgam8/a8-sidecar:0.1
+  image: amalgam8/a8-sidecar:0.1-alpine
```
Comment (rshriram) [inline docker/gateway.yaml]: can we not change this file yet?

**Pull request merged**